src/dune_pkg/lock_dir.ml

```diff
@@ -99,17 +142,67 @@ module Pkg = struct
            { Pkg_info.name; version; dev; source }
          in
          { build_command; deps; install_command; info; exported_env; lock_dir }
+
+  let encode
+      { build_command
+      ; install_command
+      ; deps
+      ; info = { Pkg_info.name = _; version; dev; source }
+      ; lock_dir = _
+      ; exported_env
+      } =
+    let open Dune_lang.Encoder in
+    record_fields
+      [ field Fields.version string version
+      ; field_o Fields.install Dune_lang.Action.encode install_command
+      ; field_o Fields.build Dune_lang.Action.encode build_command
+      ; field_l Fields.deps Package_name.encode deps
+      ; field_o Fields.source Source.encode source
+      ; field_b Fields.dev dev
+      ; field_l Fields.exported_env Env_update.encode exported_env
+      ]
 end
 
 type t =
   { version : Syntax.Version.t
   ; packages : Pkg.t Package_name.Map.t
   }
 
+let create_latest_version packages =
+  let version = Syntax.greatest_supported_version Dune_lang.Pkg.syntax in
+  { version; packages }
+
 let path = Path.Source.(relative root "dune.lock")
 
 let metadata = "lock.dune"
 
 module Metadata = Dune_sexp.Versioned_file.Make (Unit)
 
 let () = Metadata.Lang.register Dune_lang.Pkg.syntax ()
+
+let encode_metadata t =
+  let open Dune_lang.Encoder in
+  list sexp
+    [ string "lang"
+    ; string (Syntax.name Dune_lang.Pkg.syntax)
+    ; Dune_lang.Syntax.Version.encode t.version
+    ]
+
+let file_contents_by_path t =
+  (metadata, [ encode_metadata t ])
+  :: (Package_name.Map.to_list t.packages
+     |> List.map ~f:(fun (name, pkg) ->
+            (Package_name.to_string name, Pkg.encode pkg)))
+
+let write_disk ~lock_dir_path t =
+  let lock_dir_path = Path.source lock_dir_path in
+  Path.rm_rf lock_dir_path;
```

Comment:
Before deleting, how about we make sure this is a real lock file directory if it exists to avoid unfortunate accidents. For example, we check that it has a `dune.lock` file.

Reply:
ooh good idea! Will implement